include/deal.II/matrix_free/fe_point_evaluation.h

```diff
@@ -853,73 +846,42 @@ FEPointEvaluation<n_components, dim, spacedim, Number>::reinit(
   const typename Triangulation<dim, spacedim>::cell_iterator &cell,
   const ArrayView<const Point<dim>> &                         unit_points)
 {
-  // If using the fast path, we need to precompute the mapping data.
-  if (!poly.empty())
-    internal::FEPointEvaluation::compute_mapping_data_for_generic_points(
-      *mapping, cell, unit_points, update_flags_mapping, mapping_data);
-
-  // then call the other version of this function with the precomputed data
-  reinit(cell, unit_points, mapping_data);
-}
-
+  // reinit is only allowed for mapping computation on the fly
+  AssertThrow(precomputed_mapping == false, ExcNotImplemented());
 
+  if (fast_path)
+    {
+      mapping_info_on_the_fly =
+        std::make_unique<NonMatching::MappingInfo<dim, spacedim>>(*mapping,
+                                                                  update_flags);
```

Comment:
Regarding my comment above, I see now that you construct it here in `reinit`. Wouldn't it make sense to construct it in the constructor directly, as this part is always the same on different cells, and only call the `reinit` function of `MappingInfo` here? This would also be more in line with the previous workings of this class, avoiding memory re-allocations upon every `reinit`.

Reply:
Yes it makes more sense to construct it in the constructor! Before it was only constructed in case we use the fast path. I tried to separate concerns of mapping computation and shape function evaluation in commit https://github.com/dealii/dealii/pull/13835/commits/aad0799e9685b70358bc3a7883cea1c02d716d32 . Let me know what you think about this!